Add support for cloudfoundry CUPS.

toolness opened the pull request:
This fixes #306.

The `load_cups_from_vcap_services()` function was largely inspired by [python-dotenv](https://github.com/theskumar/python-dotenv#readme).

The documentation is largely cribbed from [team-api-server](https://github.com/18F/team-api-server/blob/master/README.md#on-cloudgov)'s.

jseppi commented on deploy.md:
```diff
@@ -23,6 +23,41 @@ To start, target the org and space you want to work with. For example, if you wa
+CUPS called `calc-env`, provide 'credentials' to it, and link it to the
+application instance.
+
+First, create a file called `credentials.json` with all the configuration
```
Should probably add `credentials.json` to `.gitignore`

jseppi commented:
I noticed some very minor pep8 issues in the new files (trailing spaces in tests.py and only one blank line before the `def` in settings_utils.py).

Other than that 👍

toolness commented:
Good catches! Just fixed them and also added `flake8` to `requirements.txt` so we can do linting as part of the test suite in the future, and so it's in the virtualenv/docker container of devs automatically.

jseppi commented:
👍
